syntax = "proto3";

message RequisicaoInsercao {
  int32 chave = 1;
  string valor = 2;
}

message RespostaInsercao {
  int32 status = 1;
}

message RequisicaoConsulta {
  int32 chave = 1;
}

message RespostaConsulta {
  string resultado = 1;
}

message RequisicaoAtivacao {
  string servico = 1;
}

message RespostaAtivacao {
  int32 status = 1;
}

message RequisicaoTermino {}

message RespostaTermino {
  int32 status = 1;
}

service ServerPares {
  rpc Insere(RequisicaoInsercao) returns(RespostaInsercao) {}
  rpc Consulta(RequisicaoConsulta) returns(RespostaConsulta) {}
  rpc Ativa(RequisicaoAtivacao) returns(RespostaAtivacao) {}
  rpc Termina(RequisicaoTermino) returns(RespostaTermino) {}
}
